packages/core/core/src/requests/EntryRequest.js

```diff
@@ -1,16 +1,60 @@
-// @flow
+// @flow strict-local
 import type {FileSystem} from '@parcel/fs';
 import type {FilePath, File} from '@parcel/types';
-import type {Entry, ParcelOptions} from './types';
-import path from 'path';
+import type {StaticRunOpts} from '../RequestTracker';
+import type {Entry, ParcelOptions} from '../types';
+
 import {isGlob, glob} from '@parcel/utils';
+import path from 'path';
+
+type RunOpts = {|
+  input: FilePath,
+  ...StaticRunOpts,
+|};
+
+export type EntryRequest = {|
+  id: string,
+  +type: 'entry_request',
+  run: RunOpts => Promise<EntryResult>,
+  input: FilePath,
+|};
 
 export type EntryResult = {|
   entries: Array<Entry>,
   files: Array<File>,
 |};
 
-export class EntryResolver {
+const type = 'entry_request';
+
+export default function createEntryRequest(input: FilePath) {
+  return {
+    id: `${type}:${input}`,
+    type,
+    run,
+    input,
+  };
+}
+
+async function run({input, api, options}: RunOpts): Promise<EntryResult> {
+  let entryResolver = new EntryResolver(options);
+  let result = await entryResolver.resolveEntry(input);
+
+  // Connect files like package.json that affect the entry
+  // resolution so we invalidate when they change.
+  for (let file of result.files) {
+    api.invalidateOnFileUpdate(file.filePath);
+  }
+
+  // If the entry specifier is a glob, add a glob node so
+  // we invalidate when a new file matches.
+  if (isGlob(input)) {
+    api.invalidateOnFileCreate(input);
```

Comment:
We should probably invalidate on delete here as well. Right now we get an error if you delete an entry, and you have to clear the entire cache directory to get around it.

Reply:
Okay, I added it above in the for loop since there we have the actual entry file and not the input. Is there an issue number I should refer to, so it can be closed out?